drivers/spi/spi_andes_atcspi200.c

```diff
@@ -368,28 +829,104 @@ static void spi_atcspi200_irq_handler(void *arg)
 				}
 
 			} else if (!spi_context_rx_on(ctx)) {
-				CLR_MASK(SPI_INTEN(dev), IEN_RX_FIFO_MSK);
+				sys_clear_bits(SPI_INTEN(cfg->base), IEN_RX_FIFO_MSK);
 			}
 
 			spi_context_update_rx(ctx, dfs, 1);
 		}
-		sys_write32(INTST_RX_FIFO_INT_MSK, SPI_INTST(dev));
+		sys_write32(INTST_RX_FIFO_INT_MSK, SPI_INTST(cfg->base));
 	}
 
 	if (intr_status & INTST_END_INT_MSK) {
 
 		/* Clear end interrupt */
-		sys_write32(INTST_END_INT_MSK, SPI_INTST(dev));
+		sys_write32(INTST_END_INT_MSK, SPI_INTST(cfg->base));
 
 		/* Disable all SPI interrupts */
-		sys_write32(0, SPI_INTEN(dev));
+		sys_write32(0, SPI_INTEN(cfg->base));
+
+#ifdef CONFIG_ANDES_SPI_DMA_MODE
+		if ((data->dma_tx.dma_dev != NULL)
+		 && data->dma_tx.dma_cfg.source_chaining_en) {
+
+			spi_tx_dma_disable(dev);
+			dma_stop(data->dma_tx.dma_dev, data->dma_tx.channel);
+			data->dma_tx.block_idx = 0;
+			data->dma_tx.dma_blk_cfg.next_block = NULL;
+		}
+
+		if ((data->dma_rx.dma_dev != NULL)
+		 && data->dma_rx.dma_cfg.source_chaining_en) {
```

Comment:
```suggestion
		if ((data->dma_rx.dma_dev != NULL) && data->dma_rx.dma_cfg.source_chaining_en) {
```

Reply:
Thanks for your suggestion!
We have update the source code.